src/Forms/quadraticform.jl

```diff
@@ -0,0 +1,70 @@
+struct QuadraticForm{V,KetMap,BraMap,OperatorMap,VectorSpaceMap,LinkSpaceMap} <:
+       AbstractForm{V,KetMap,BraMap,OperatorMap,VectorSpaceMap}
+  tn::ITensorNetwork{V}
+  bra_vs_map::BraMap
+  ket_vs_map::KetMap
+  operator_vs_map::OperatorMap
+  dual_space_map::VectorSpaceMap
+  link_space_map::LinkSpaceMap
+end
+
+#Needed for implementation
+bra_map(qf::QuadraticForm) = qf.bra_vs_map
+ket_map(qf::QuadraticForm) = qf.ket_vs_map
+operator_map(qf::QuadraticForm) = qf.operator_vs_map
+space_map(qf::QuadraticForm) = qf.dual_space_map
+tensornetwork(qf::QuadraticForm) = qf.tn
+data_graph_type(::Type{<:QuadraticForm}) = data_graph_type(tensornetwork(qf))
+data_graph(qf::QuadraticForm) = data_graph(tensornetwork(qf))
+
+function QuadraticForm(
+  ket::ITensorNetwork, operator::ITensorNetwork; link_space_map=default_index_map, kwargs...
+)
+  bra = link_space_map(dag(ket); sites=[])
+  return Form(bra, operator, ket; form_type="Quadratic", link_space_map, kwargs...)
+end
+
+function QuadraticForm(
+  ket::ITensorNetwork;
+  dual_space_map=default_index_map,
+  tno_constructor=default_tno_constructor,
+  kwargs...,
+)
+  s = siteinds(ket)
+  operator_space = union_all_inds(s, dual_space_map(s; links=[]))
+  operator = tno_constructor(operator_space)
+  return QuadraticForm(ket, operator; kwargs...)
+end
+
+function bra_ket_vertices(qf::QuadraticForm, state_vertices::Vector)
+  ket_vertices = [ket_map(qf)[sv] for sv in state_vertices]
+  bra_vertices = [bra_map(qf)[sv] for sv in state_vertices]
+  return bra_vertices, ket_vertices
+end
+
+function gradient(
```

Comment:
I think it should be called `derivative`, since it is just taking the derivative with respect to one group of vertices. `gradient` would be computing the derivative for multiple sets of inputs (say, multiple derivatives, one for each input). See for example https://docs.juliahub.com/General/AbstractDifferentiation/stable/user_guide/#AbstractDifferentiation-backends.

Reply:
Actually that link isn't so helpful, I can't really make sense of how `AbstractDifferentiation.jl` distinguishes between `derivative` and `gradient`.

For our case, the way I see it using `derivative` could help distinguish the potential ambiguity that:
```julia
gradient(f, [v1, v2, v3])
```
could refer to either:
```julia
[∂ᵥ₁f, ∂ᵥ₂f, ∂ᵥ₃f]
```
i.e. gradient with respect to the three specified vertices, or:
```julia
∂ᵥ₁∂ᵥ₂∂ᵥ₃f
```
i.e. a single 3rd order derivative, which is closer to what we are looking for with this operation. So I was thinking using `derivative(f, [v1, v2, v3])` would make it clearer that we mean the latter, and we could leave `gradient` to represent the former.

Reply:
Okay that makes sense, I have moved to using `derivative` instead